src/poetry/factory.py

```diff
@@ -110,23 +112,23 @@ def get_package(cls, name: str, version: str) -> ProjectPackage:
         return ProjectPackage(name, version, version)
 
     @classmethod
-    def configure_sources(
+    def create_pool(
         cls,
-        poetry: Poetry,
-        sources: list[dict[str, str]],
         config: Config,
-        io: IO,
+        io: IO | None = None,
         disable_cache: bool = False,
-    ) -> None:
-        if disable_cache:
-            logger.debug("Disabling source caches")
+    ) -> Pool:
+        from poetry.repositories.pool import Pool
 
-        for source in sources:
-            repository = cls.create_package_source(
-                source, config, disable_cache=disable_cache
-            )
-            is_default = bool(source.get("default", False))
-            is_secondary = bool(source.get("secondary", False))
+        if io is None:
+            io = NullIO()
+
+        pool = Pool()
+
+        for source in config.get("sources", {}).values():
+            repository = cls.create_package_source(source, config)
```

Comment:
```suggestion
            repository = cls.create_package_source(
                source, config, disable_cache=disable_cache
            )
```
It seems that got lost, too.